command/recorderd/config_reader.go

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2014-2019 Bitmark Inc.
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/bitmark-inc/logger"
+)
+
+type ConfigReader interface {
+	initialise(string)
+	optimalThreadCount() uint32
+	setCalendar(JobCalendar)
+	refresh() error
+	getConfig() (*Configuration, error)
+	setLog(*logger.L) error
+	updatePeriodic()
```

Comment:
suggested naming: `periodicallyUpdate`.

Reply:
I try to use all verb + noun, which is more consistent in these methods

Reply:
Isn't `Periodic` an adj...?

Reply:
yeah, so maybe `updatePeriodically` would be better fitting naming